addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -506,10 +506,15 @@ export class WebsitePreview extends Component {
         // If the iframe is currently displaying an XML file, the body does not
         // exist, so we do not replace the iframefallback content.
         // The iframefallback is hidden in test mode
-        if (!this.websiteContext.edition && this.iframe.el.contentDocument.body && this.iframefallback.el) {
-            this.iframefallback.el.contentDocument.body.replaceWith(this.iframe.el.contentDocument.body.cloneNode(true));
+        const websiteDoc = this.iframe.el && this.iframe.el.contentDocument;
+        const fallbackDoc = this.iframefallback.el && this.iframefallback.el.contentDocument;
+        if (!this.websiteContext.edition && websiteDoc && fallbackDoc) {
+            for (const el of websiteDoc.head.querySelectorAll("link[rel='stylesheet'], style")) {
+                fallbackDoc.head.appendChild(el.cloneNode(true))
```

Comment:
Clear the fallback iframe’s head before appending the cloned styles so we don’t keep stacking duplicate link/style tags on each reload 😉